File: fontend-main/src/app/core/services/skin-type.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, catchError, map, of } from 'rxjs';
import { HttpClientService } from './httpClient.service';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SkinTypeService {
  httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json',
    })
  };
  constructor(private http: HttpClient, private httpService: HttpClientService){}

  //#region CRUD
  search(): Observable<any> {
    return this.http.get(`${environment.baseUrl}${environment.basePath}/skin-type`);
  }

  create(data: any): Observable<any> {
    data = JSON.parse(data);
    return this.http.post(`${environment.baseUrl}${environment.basePath}/skin-type`, data);
  }

  update(id: string, data: any): Observable<any> {
    const jsonData = JSON.stringify(data);
    return this.http.put(`${environment.baseUrl}${environment.basePath}/skin-type/` +  id, jsonData, this.httpOptions);
  }

  getById(id: string): Observable<any> {
    return this.http.get(`${environment.baseUrl}${environment.basePath}/skin-type/` + id);
  }

  delete(id: string): Observable<any> {
    return this.http.delete(`${environment.baseUrl}${environment.basePath}/skin-type/` + id);
  }
  //#endregion
}

File: backend-main/src/main/java/com/cosmetic/gg/service/attribute/SkinTypeService.java
package com.cosmetic.gg.service.attribute;

import java.util.List;

import com.cosmetic.gg.common.model.Error;
import com.cosmetic.gg.entity.attribute.SkinType;

public interface SkinTypeService {
	
	List<SkinType> search();

	List<Error> validator(SkinType skinType);
	
	SkinType create(SkinType skinType);
	
	SkinType update(SkinType skinType);
	
	SkinType getById(String id);
	
	Error delete(String id);
}
